### PR TITLE
Update version bump tool

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Update Version Bump tool to allow local/manual version bumps
- Run tool for Azure.Core to bump up version

### CODE REVIEW

File: sdk/core/Azure.Core/src/Azure.Core.csproj [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-    <Version>1.0.2</Version>
+    <Version>1.1.0-preview.1</Version>
     <PackageTags>Microsoft Azure Client Pipeline</PackageTags>
     <Nullable>enable</Nullable>
     <DefineConstants>$(DefineConstants);AZURE_NULLABLE</DefineConstants>
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <EnableClientSdkAnalyzers>false</EnableClientSdkAnalyzers>
   </PropertyGroup>
-
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/Azure.Core.csproj]: Can we avoid tool removing the formatting?

Comment (weshaggard) [reply sdk/core/Azure.Core/src/Azure.Core.csproj]: I think you can set PreserveWhitespace to get this to stick around, see https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmldocument.preservewhitespace

File: eng/Version-Increment.ps1 [hunk 2]
```diff
@@ -17,85 +17,128 @@ The Name of the Package
...
+    $VersionTable = Parse-Version($NewVersionString)
+    if ($VersionTable['pretag'] -eq '')
+    {
+        $NewPackageVersion = "{0}.{1}.{2}" -F $VersionTable['major'], $VersionTable['minor'], $VersionTable['patch']
+    }
+    else
+    {
+        $NewPackageVersion = "{0}.{1}.{2}-{3}.{4}" -F $VersionTable['major'], $VersionTable['minor'], $VersionTable['patch'], $VersionTable['pretag'], $VersionTable['prever']
+    }
+    Update-Version -ReplaceVersion $True
```
Comment (weshaggard) [inline eng/Version-Increment.ps1]: In this case I suspect most people also want to change from unreleased to stamp a date, as they will generally call this right before they release to set the final version.

File: eng/Version-Increment.ps1 [hunk 3]
```diff
@@ -17,85 +17,128 @@ The Name of the Package
 .PARAMETER PackageDirName
 Used in the case where the package directory name is different from the package name. e.g in cognitiveservice packages
 
-.EXAMPLE
-Updating package version for core
+.PARAMETER NewVersionString
+Use this to overide version incement logic and set a version specified by this parameter
+
 
+.EXAMPLE
```
Comment (weshaggard) [inline eng/Version-Increment.ps1]: Can you please add some docs about this script in Versioning.md file?

File: eng/Version-Increment.ps1 [hunk 4]
```diff
@@ -17,85 +17,128 @@ The Name of the Package
...
+Updating package version for Azure.Core
 Version-Increment.ps1 -RepoRoot "C:\Git\azure-sdk-for-net" -ServiceDirectory core -PackageName Azure.Core
 
+Updating package version for Azure.Core with a specified verion
+Version-Increment.ps1 -RepoRoot "C:\Git\azure-sdk-for-net" -ServiceDirectory core -PackageName Azure.Core -NewVersionString 2.0.5
```
Comment (weshaggard) [inline eng/Version-Increment.ps1]: I think I would eliminate the RepoRoot from the samples to simplify them.

File: eng/Version-Increment.ps1 [hunk 5]
```diff
@@ -17,85 +17,128 @@ The Name of the Package
 .PARAMETER PackageDirName
 Used in the case where the package directory name is different from the package name. e.g in cognitiveservice packages
 
-.EXAMPLE
-Updating package version for core
+.PARAMETER NewVersionString
```
Comment (weshaggard) [inline eng/Version-Increment.ps1]: minor but now that this also supports setting a version I'm not sure "Version-Increment" is the best name for the script. Perhaps having it called "Update-Version" with an `-increment` option which is mutually exclusive with the `-newVersion` option.

File: eng/Version-Increment.ps1 (hunk 3 above)
Comment (chidozieononiwu) [reply eng/Version-Increment.ps1]: Done

File: eng/Version-Increment.ps1 (hunk 2 above)
Comment (chidozieononiwu) [reply eng/Version-Increment.ps1]: Updated the call to `Update-Changelog.ps1` script to handles this

**Review by pakrym — APPROVED** [review]
From Azure.Core point of view

File: eng/Update-PkgVersion.ps1 [hunk 6]
```diff
@@ -0,0 +1,145 @@
+<#
```
Comment (weshaggard) [inline eng/Update-PkgVersion.ps1]: Remember this filename change will need a matching change in the release pipeline, or if you want to do safer you can make the Version-Increment script a simple wrapper around this one.

Comment (chidozieononiwu) [reply eng/Update-PkgVersion.ps1]: I have a PR out for the name change.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**